### ISSUE TITLE
Unable to prevent Clang headers from being used

### ISSUE THREAD

**Issue by wmmc88** [rust-lang/rust-bindgen#2842]
All versions of LLVM 18 have a bug where the `__prefetch` intrinsic has the wrong definition. This bug was fixed in https://github.com/llvm/llvm-project/pull/93235, but will not be released until LLVM 19 comes out.

When `bindgen` invokes `clang` to parse the headers, it looks like `clang` is actually using its bugged `intrin.h` which brings in the invalid signature for `__prefetch`. This leads to the failure to generate bindings for any header that pulls in msvc's `intrin.h` on `aarch64`. I attempted to use `clang`'s `-fno-builtin` arg to prevent `bindgen` from pulling in the wrong definition of `__prefetch`, but it seems to not have any effect here.

### Bindgen Invocation

```Rust
use std::env;
use std::path::PathBuf;

fn main() {
    let bindings = bindgen::Builder::default()
        .header_contents(
            "input.h",
            r#"
void __prefetch(const void *);"#,
        )
        .parse_callbacks(Box::new(bindgen::CargoCallbacks::new()))
        .clang_args([
            "-fms-compatibility",
            "-fms-extensions",
            "-fdelayed-template-parsing",
            "-fno-builtin", // attempt to prevent intrin.h from clang from being used (not working)
            "-fno-builtin-__prefetch" // another not working attempt
        ])
        .generate()
        .expect("Unable to generate bindings");

    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
    bindings
        .write_to_file(out_path.join("bindings.rs"))
        .expect("Couldn't write bindings!");
}

```

### Actual Results

```
cargo build --target aarch64-pc-windows-msvc
   Compiling bindgen-test v0.1.0 (D:\git-repos\bindgen-test)
error: failed to run custom build command for `bindgen-test v0.1.0 (D:\git-repos\bindgen-test)`

Caused by:
  process didn't exit successfully: `D:\git-repos\bindgen-test\target\debug\build\bindgen-test-28575ca189a67a38\build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64-pc-windows-msvc
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64_pc_windows_msvc
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS

  --- stderr
  D:\git-repos\bindgen-test\input.h:2:6: error: conflicting types for '__prefetch'
  D:\git-repos\bindgen-test\input.h:2:6: note: '__prefetch' is a builtin with type 'void (void *)'
  thread 'main' panicked at build.rs:23:10:
  Unable to generate bindings: ClangDiagnostic("D:\\git-repos\\bindgen-test\\input.h:2:6: error: conflicting types for '__prefetch'\n")
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace```


### Expected Results

Should compile sucessfully

**Comment by workingjubilee** [comment]
why is clang not issuing a patch release?

**Comment by workingjubilee** [comment]
> Unfortunately, this can't be backported (there are no more 18.x releases, and this change is a breaking change), so I'll see if I can get a workaround added on MSVC's side for Clang 18.

ah.

**Comment by pvdrz** [comment]
@workingjubilee @wmmc88 is there something we can do from `bindgen`'s side to mitigate this?

**Comment by pvdrz** [comment]
it depends, if you can filter it using `allowlist_file` or any other allowlist method maybe we can do something about it.

**Comment by jdoubleu** [comment]
> This bug was fixed in https://github.com/llvm/llvm-project/pull/93235, but will not be released until LLVM 19 comes out.

By now, LLVM 19 has already been published.

Is this still an issue?

**Comment by wmmc88** [comment]
> > This bug was fixed in https://github.com/llvm/llvm-project/pull/93235, but will not be released until LLVM 19 comes out.
> 
> By now, LLVM 19 has already been published.
> 
> Is this still an issue?

The bug in llvm that original motivated my ask is fixed, but the original ask of bindgen having any sort of knob to prevent clang headers from being pulled in is still unaddressed.

**Comment by mbostwick** [comment]
This bug breaks wdk-sys .. and I just lost several hours to resolving that it was not a result of the [WDK version](https://learn.microsoft.com/en-us/windows-hardware/drivers/other-wdk-downloads) but the LLVM version as using the latest version causes a compile error that reports: `attempt to compute 1_usize - 352_usize, which would overflow` as the resolve error makes the size 1_ not match the header size ..

**Comment by mbostwick** [comment]
> > This bug was fixed in [llvm/llvm-project#93235](https://github.com/llvm/llvm-project/pull/93235), but will not be released until LLVM 19 comes out.
> 
> By now, LLVM 19 has already been published.
> 
> Is this still an issue?

yes, and things are getting out of sync making this problem get worse, with the pending merge requests for this all cleared.  Any chance we can add a help wanted or good first issue label somewhere ? I was just aiming to do a simple driver, and didn't expect to be concerned about my LLVM versions between projects ...

Here is a small MD file on steps to reproduce in hopes it will make the issue easier to pick up:

https://gist.github.com/mbostwick/b302889cd97670abcb2186e2b7d629a2

The nice thing about this is its just simple change over time and making things compile between the versions (so it makes for a good issue for people that want to get into LLVM and rust-bindgen ).
